scvi/module/_utils.py

```diff
@@ -41,3 +41,16 @@ def batch(batch_size, label):
 
     batch_size = x.size(0)
     return torch.cat([batch(batch_size, i) for i in range(y_dim)])
+
+
+def masked_softmax(weights, mask, dim=-1, eps=1e-30):
```

Comment:
can you add a proper docstring and typing here?

Reply:
Done, first line is summary.